Module 5/Ex0ToEx5 V2/Sources/OfficeBlock.cpp

```diff
@@ -22,13 +28,41 @@ void OfficeBlock::SetExecutor(std::unique_ptr<Bureaucrat> executing)
     _executing = std::move(executing);
 }
 
-void OfficeBlock::DoBureaucracy(FormType formType, const std::string &target) const
+void OfficeBlock::InitializeFormCreationMap()
+{
+    _formCreationMap["mutant pig termination"] = &OfficeBlock::CreateGenericForm;
+}
+
+std::unique_ptr<Form> OfficeBlock::CreateGenericForm(const std::string &formName, const std::string &target)

```

Comment:
The Office Block is not responsible for the creation of a form, the intern is.
This method should be moved to the intern.

Moreover, with your method, no new type of form is created. The different types of form are already known by the intern, so a MutantPigTerminationForm class would be more appropriate than the generic form